lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -443,146 +517,243 @@ def create_autoscaling_group(connection, module):
     health_check_type = module.params.get('health_check_type')
     default_cooldown = module.params.get('default_cooldown')
     wait_for_instances = module.params.get('wait_for_instances')
-    as_groups = connection.get_all_groups(names=[group_name])
+    as_groups = connection.describe_auto_scaling_groups(AutoScalingGroupNames=[group_name])
     wait_timeout = module.params.get('wait_timeout')
     termination_policies = module.params.get('termination_policies')
     notification_topic = module.params.get('notification_topic')
     notification_types = module.params.get('notification_types')
 
     if not vpc_zone_identifier and not availability_zones:
-        region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+        region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
         try:
-            ec2_connection = connect_to_aws(boto.ec2, region, **aws_connect_params)
-        except (boto.exception.NoAuthHandlerFound, AnsibleAWSError) as e:
+            ec2_connection = boto3_conn(module,
+                                        conn_type='client',
+                                        resource='ec2',
+                                        region=region,
+                                        endpoint=ec2_url,
+                                        **aws_connect_params)
+        except (botocore.exceptions.NoCredentialsError, AnsibleAWSError) as e:
             module.fail_json(msg=str(e))
     elif vpc_zone_identifier:
         vpc_zone_identifier = ','.join(vpc_zone_identifier)
 
     asg_tags = []
     for tag in set_tags:
-        for k,v in tag.items():
+        for k,v in iteritems(tag):
```

Comment:
I wondered about this too. I think he chose this because iterating over the key/values in a dict has changed in python3 and module_utils.six checks for the python version so the behavior is consistent. It seems to behave appropriately. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/six.py#L367
Do you think this might end up being a problem?